authentication/managing_cloud_provider_credentials/cco-mode-sts.adoc

```diff
@@ -0,0 +1,734 @@
+----
+
+.. Calculate the fingerprint for the certificate at the root of the chain:
++
+[source,terminal]
+----
+$ export BUCKET_FINGERPRINT=$(openssl x509 -in cert<number>.pem -fingerprint -noout | sed -e 's/.*Fingerprint=//' -e 's/://g')
+----
++
+Where `<number>` is the highest number in the files that were saved. For example, if two is the highest number in the files that were saved, use `cert2.pem`.
+
+.. Create the identity provider:
++
+[source,terminal]
+----
+$ aws iam create-open-id-connect-provider --url $OPENID_BUCKET_URL --thumbprint-list $BUCKET_FINGERPRINT
+----
+
+.. Save the returned ARN of the newly created identity provider to `$AWS_IAM_OPENID_ARN`.
+
+. Generate IAM roles:
+
+.. Locate all `CredentialsRequest` CRs in this release image that target the cloud you are deploying on:
++
+[source,terminal]
+----
+$ oc adm release extract quay.io/openshift-release-dev/ocp-release:4.y.z-x86_64 --credentials-requests --cloud=aws
+----
+
+.. For each `CredentialsRequest` CR, create an IAM role that grants the necessary permissions and establishes a trust relationship that trusts the  identity provider previously created.
++
+For example, for the openshift-machine-api-operator `CredentialsRequest` CR in `0000_30_machine-api-operator_00_credentials-request.yaml`, create an IAM role that allows an identity from the created OIDC provider created for the cluster:
++
+[source,terminal]
+----
+$ aws iam get-role --role-name openshift-machine-api-aws-cloud-credentials
+----
++
+[source,json]
+----
+{
+    "Role": {
+        "Path": "/",
+        "RoleName": "openshift-machine-api-aws-cloud-credentials",
+        "RoleId": "ARSOMEROLEID",
+        "Arn": "arn:aws:iam::123456789012:role/openshift-machine-api-aws-cloud-credentials",
+        "CreateDate": "2021-01-06T15:54:13Z",
+        "AssumeRolePolicyDocument": {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Principal": {
+                        "Federated": "$AWS_IAM_OPENID_ARN"
+                    },
+                    "Action": "sts:AssumeRoleWithWebIdentity",
+                    "Condition": {
+                        "StringEquals": {
+                            "s3.<aws-region>.amazonaws.com/$BUCKET_NAME:aud": "openshift"
+                        }
+                    }
+                }
+            ]
+        },
+        "Description": "OpenShift role for openshift-machine-api/aws-cloud-credentials",
+        "MaxSessionDuration": 3600,
+        "RoleLastUsed": {
+            "LastUsedDate": "2021-02-03T02:51:24Z",
+            "Region": "<aws-region>"
+        }
+    }
+}
+----
+
+.. To further restrict the role such that only specific cluster `ServiceAccount` objects can assume the role, update the `.Role.AssumeRolePolicyDocument.Statement[].Condition` field to the specific `ServiceAccount` object for each component.
```

Comment:
Are these steps similar to Step 2 in pre-flight? I like the _modify the trust relationship_ wording up there.

Reply:
I think that's right - updated to make the construction the same.

Reply:
Changed to make more similar